src/genotyper.cc

```diff
@@ -664,6 +977,19 @@ Status genotype_site(const genotyper_config& cfg, MetadataCache& cache, BCFData&
         update_orig_calls_for_loss(cfg, records, bcf_nsamples, dataset_header.get(),
                                    sample_mapping, loss_trackers);
 
+        // Update format fields
+        for (auto& record : records) {
+            vector<int> allele_mapping(record->n_allele, -1);
+            vector<bool> deletion_allele(record->n_allele, false); // which alleles are deletions
+
+            S(find_allele_mapping(site, record.get(), allele_mapping, deletion_allele))
```

Comment:
Yes... this was a bit annoying... it is possible to calculate allele_mapping only once in `genotype_site`, but that means we need to pass it to `find_variant_records` so that the record <-> allele_mapping (for record) pairing is preserved for the variant records, then pass the selected variant records with the corresponding allele_mapping into `translate_genotype`, which seemed even more convoluted in my opinion...